packages/@react-spectrum/datepicker/stories/DateRangePicker.stories.tsx

```diff
@@ -86,6 +87,13 @@ storiesOf('Date and Time/DateRangePicker', module)
     'minDate: 2010/1/1, maxDate: 2020/1/1',
     () => render({minValue: new CalendarDate(2010, 1, 1), maxValue: new CalendarDate(2020, 1, 1)})
   )
+  .add(
+    'multiple disabled intervals',
+    () => render({isDateDisabled: (date: DateValue) => {
+      const disabledIntervals = [[today(getLocalTimeZone()), today(getLocalTimeZone()).add({weeks: 1})], [today(getLocalTimeZone()).add({weeks: 2}), today(getLocalTimeZone()).add({weeks: 3})]];
+      return disabledIntervals.some((interval) => date.compare(interval[0]) > 0 && date.compare(interval[1]) < 0);
+    }})
+  )
```

Comment:
IMO kinda feels odd to be able to choose a date range that contains disabled dates. I was thinking of a use case for choosing a Airbnb booking date range and it would be odd being able to choose a range that encompasses days where said room is booked. Perhaps the behavior should be changed for DateRangePicker or maybe the prop should be further described as only blocking start/end date selection? Maybe selecting one date should trigger an update to the calendar dates to make it impossible to select a range that includes a disabled date?

Reply:
Hmmm you could compare this to a Workday time off request though. In that case, weekends and holidays are disabled dates. But you can select a range of several weeks that includes disabled dates, it'll just ignore those dates.

I'm not sure how much sense the returned value makes though if you're doing that? I guess it still makes sense, it's a single range and the user was informed that certain dates within it are disabled, and the app already knows how to calculate those, so it could be applied however the application likes.

Reply:
IMO it is more flexible to allow selected intervals, for those reasons:
- Validation can be done outside the `DateRangePicker` You have access to the specified disabled intervals, so if the selected interval overlaps, you can warn the user.
- There are cases when you want to be able to select disabled ranges, as it would be way more painful for the user to select multiple intervals that are not disabled (ex: selecting 4 weeks without the weekends, instead of the whole month). Again, the logic for excluding the disabled intervals can be done outside of the date picker.